src/main/java/frc/robot/XBoxOneController.java

```diff
@@ -0,0 +1,285 @@
+/*
+ * Copyright (c) 2014-2017, FRC3161
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without modification,
+ * are permitted provided that the following conditions are met:
+ *
+ * * Redistributions of source code must retain the above copyright notice, this
+ *   list of conditions and the following disclaimer.
+ *
+ * * Redistributions in binary form must reproduce the above copyright notice, this
+ *   list of conditions and the following disclaimer in the documentation and/or
+ *   other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+package frc.robot;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Objects;
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Consumer;
+import java.util.function.Function;
+
+import ca.team3161.lib.utils.Utils;
+import ca.team3161.lib.utils.controls.AbstractController;
+import ca.team3161.lib.utils.controls.LinearJoystickMode;
+import ca.team3161.lib.utils.controls.Gamepad;
+
+/**
+ * A Gamepad implementation describing the Logitech DualAction gamepad.
+ */
+public class XBoxOneController extends AbstractController {
+
+    /**
+     * {@inheritDoc}.
+     */
+    public enum XBoxOneControl implements Control {
+        LEFT_STICK(0),
+        RIGHT_STICK(1),
+        LEFT_TRIGGER(7),
+        RIGHT_TRIGGER(8);
```

Comment:
What happened to the direct accessors you said you would add for getting the trigger axis values?